addons/mass_mailing/controllers/main.py

```diff
@@ -195,3 +195,13 @@ def send_feedback(self, mailing_id, res_id, email, feedback, token):
                 record.sudo().message_post(body=_("Feedback from %(email)s: %(feedback)s", email=email, feedback=feedback))
             return bool(records)
         return 'error'
+
+    @http.route('/mailing/turnoff_report', type='http', website=True, auth='user')
```

Comment:
Why not auth=public ? That way even if not logged, with a right token this could be turned off. It means that receiving annoying spam deactivated even if I forgot my login/password, and I will not land on NotFound page without knowing why.

We could get a token and a user_id from parameters and act accordingly (a bit like in digest). Idea would be
  * I have a token and a user_id -> check token is correct, grant access based on user_id
  * no token or no user_id -> check based on current user rights (may be public)

Reply:
Sounds good! Regarding the second point, I have been taught by @std-odoo not to have that kind of routes without a token as someone could send an email with an image with this route as a src and it would unsubscribe all the users (not sure why someone would do that...)

Reply:
Indeed. Anyway people will receive links with token and user_id so no need to support the other global route. Config settings are there for you anyway. Good remark.

Reply:
Yes I agree to make the route public :) 
So the check of `request.env['res.users'].has_group('mass_mailing.group_mass_mailing_user')` must be adapted to check the id of the given user (and not the current one) and we should give the user_id to the route (can be a different GET parameter, no need to merge the token and the user_id I think)

Reply:
@fbarbe00 Ping me when you did it so I double check :)